src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/ClusterAwareWriterFailoverHandler.java

```diff
@@ -249,7 +255,10 @@ && isCurrentHostWriter(latestTopology)) {
               return new WriterFailoverResult(true, false, latestTopology, conn, "TaskA");
             }
           } catch (SQLException exception) {
-            // ignore
+            // Propagate exceptions that are not caused by network errors.
+            if (!exception.getSQLState().startsWith("08")) {
```

Comment:
I'd extract this condition into a separate (static?) function IsNetworkException(e)

Reply:
Makes sense